koji_containerbuild/plugins/builder_containerbuild.py

```diff
@@ -945,3 +954,271 @@ def handler(self, src, target, opts=None):
+        self.logger.debug("Started by %s", owner_info['name'])
+
+        create_build_args = {
+            'user': owner_info['name'],
+            'component': component,
+            'sources_for_koji_build_id': koji_build_id,
+            'sources_for_koji_build_nvr': koji_build_nvr,
+            'target': target_info['name'],
+            'scratch': scratch,
+            'koji_task_id': self.id,
+        }
+
+        try:
+            source_create_build_args = create_build_args.copy()
+            if signing_intent:
+                source_create_build_args['signing_intent'] = signing_intent
+
+            create_method = self.osbs().create_source_container_build
+            self.logger.debug("Starting %s with params: '%s",
+                              create_method, source_create_build_args)
+            build_response = create_method(**source_create_build_args)
+        except AttributeError:
+            raise koji.BuildError("method %s doesn't exists in osbs" % create_method)
+
+        build_id = build_response.get_build_name()
+        self.logger.debug("OSBS build id: %r", build_id)
+
+        # When builds are cancelled the builder plugin process gets SIGINT and SIGKILL
+        # If osbs has started a build it should get cancelled
+        def sigint_handler(*args, **kwargs):
+            if not build_id:
+                return
+
+            self.logger.warn("Cannot read logs, cancelling build %s", build_id)
+            self.osbs().cancel_build(build_id)
+
+        signal.signal(signal.SIGINT, sigint_handler)
+
+        self.logger.debug("Waiting for osbs build_id: %s to be scheduled.",
+                          build_id)
+        # we need to wait for kubelet to schedule the build, otherwise it's 500
+        self.osbs().wait_for_build_to_get_scheduled(build_id)
+        self.logger.debug("Build was scheduled")
+
+        osbs_logs_dir = self.resultdir()
+        koji.ensuredir(osbs_logs_dir)
+        pid = os.fork()
+        if pid:
+            try:
+                self._incremental_upload_logs(pid)
+            except koji.ActionNotAllowed:
+                pass
+        else:
+            self._osbs = None
+
+            try:
+                self._write_incremental_logs(build_id, osbs_logs_dir)
+            except Exception as error:
+                self.logger.info("Error while saving incremental logs: %s", error)
+                os._exit(1)
+            os._exit(0)
+
+        response = self.osbs().wait_for_build_to_finish(build_id)
+
+        self.logger.debug("OSBS build finished with status: %s. Build "
+                          "response: %s.", response.status,
+                          response.json)
+
+        self.logger.info("Response status: %r", response.is_succeeded())
+
+        if response.is_cancelled():
+            self.session.cancelTask(self.id)
+            raise ContainerCancelled(
+                'Image build was cancelled by OSBS, maybe by automated rebuild.'
```

Comment:
No automated rebuilds for source containers.

Reply:
changed